featurecontrol/featurecontrol.go

```diff
@@ -120,7 +120,7 @@ func NewFlags(logger *slog.Logger, features string) (Flagger, error) {
 			logger.Warn("Experimental receiver name in metrics enabled")
 		case FeatureClassicMode:
 			opts = append(opts, enableClassicMode())
-			logger.Warn("Classic mode enabled")
+			logger.Warn("UTF-8 strict mode is disabled; non-UTF-8 characters will be accepted")
```

Comment:
This doesn't quite explain what classic mode does, as it's a little more then UTF-8 strict mode being disabled. Let me take a look at the Mimir issue to see how we can help here because "classic mode" is documented in the Alertmanager UTF-8 migration guide for over 12 months now, and I don't want to create additional confusion renaming bits of it.